src/lib/encoder/test/EncoderRoundTripTest.cpp

```diff
@@ -41,11 +41,47 @@ TEST_CASE( "EncoderRoundTripTest/testFountain.Pad", "[unit]" )
 	Decoder dec(30);
 	fountain_decoder_sink<cimbar::zstd_decompressor<std::ofstream>> fds(tempdir.path(), cimbar::Config::fountain_chunk_size(30, 6, false));
 
-	unsigned bytesDecoded = dec.decode_fountain(encodedImg, fds);
+	unsigned bytesDecoded = dec.decode_fountain(encodedImg, fds, 1);
 	assertEquals( 7500, bytesDecoded );
 
 	std::string decodedContents = File(tempdir.path() / "0.626").read_all();
 	assertEquals( "hello", decodedContents );
+
+	assertEquals( 1, fds.num_done() );
+}
+
+TEST_CASE( "EncoderRoundTripTest/testFountain.SinkMismatch", "[unit]" )
+{
+	MakeTempDirectory tempdir;
+
+	std::string inputFile = tempdir.path() / "hello.txt";
+	std::string outPrefix = tempdir.path() / "encoder.fountain";
+
+	{
+		std::ofstream f(inputFile);
+		f << "hello"; // 5 bytes!
+	}
+
+	// will be padded so the fountain encoding is happy. The encoded image looks suspiciously non-random!
+	Encoder enc(30, 4, 2);
+	assertEquals( 1, enc.encode_fountain(inputFile, outPrefix) );
+
+	uint64_t hash = 0xeecc8800efce8c48;
+	std::string path = fmt::format("{}_0.png", outPrefix);
+	cv::Mat encodedImg = cv::imread(path);
+	cv::cvtColor(encodedImg, encodedImg, cv::COLOR_BGR2RGB);
+	assertEquals( hash, image_hash::average_hash(encodedImg) );
+
+	// decoder
+	Decoder dec(30);
+	// sink with a mismatched fountain_chunk_size
+	// importantly, the sink expects a *larger* chunk than we'll give it...
+	fountain_decoder_sink<cimbar::zstd_decompressor<std::ofstream>> fds(tempdir.path(), cimbar::Config::fountain_chunk_size(30, 6, true));
+
+	unsigned bytesDecoded = dec.decode_fountain(encodedImg, fds, 1);
+	assertEquals( 7500, bytesDecoded );
```

Comment:
The key here is that the fountain decode actually *succeeds*, the aggregator (the fountain_decoder_sink) just isn't prepared for it.

Reply:
... but with further testing, it looks like this test case isn't sufficient :upside_down_face:

Reply:
Error was a result of feeding bad data to wirehair -- resolved by https://github.com/sz3/libcimbar/pull/94/commits/f1e136f143ec87c8a751cdbc28d3403cd2db07d4